.github/workflows/build-and-push-cliain.yml

```diff
@@ -6,6 +6,8 @@ on:
     branches:
       - 'main'
   pull_request:
+    paths:
+      - bin/cliain/**
```

Comment:
yes, but ideally we would have also a check that `Cargo.lock`s are up-to-date - it used to be pretty common to make changes only to `aleph-client`, which actually should affect building `cliain`, which requires bumping `aleph-client` and updating `Cargo.lock` in `cliain`

not a blocker, just raising warning

Reply:
I understand, but the bulk of the time on this PR was already spent on task not related to the original issue. 
Just goes on to show there's still some work left to do on the pipelines